frontend/src/metabase-types/api/modelIndexes.ts

```diff
@@ -25,3 +25,9 @@ export type IndexedEntity = {
   name: string;
   pk_ref: FieldReference;
 };
+
+export type ModelIndexesListQuery = {
+  model_id: CardId | null;
+};
+
+export type ModelIndexesListResult = ModelIndex[];
```

Comment:
I think the shape of the response as defined in `ModelIndex` became outdated over time. If you observe the response in dev tools or using any API tool, you'll see that `generation` and `state_changed_at` don't exist.

Two new keys I see in the response are:
- `created_at` and
- `indexed_at`
(both are datetime strings)

Here's the example response:
```json
[
	{
		"value_ref": [
			"field",
			71,
			null
		],
		"schedule": "0 0 4 * * ? *",
		"pk_ref": [
			"field",
			68,
			null
		],
		"state": "indexed",
		"creator_id": 1,
		"model_id": 78,
		"id": 3,
		"indexed_at": "2024-02-15T10:12:08.806019+01:00",
		"error": null,
		"created_at": "2024-02-15T10:12:08.664025+01:00"
	}
]
```

@ranquild @kamil (and @iethree ) should this type be updated?
If this new hook proves to be working (and that's a huge if), should we maybe introduce it to `master` and update types? Also, should we remove the current conditional logic introduced [here](https://github.com/metabase/metabase/pull/37284/files#diff-24b8c743de25248f2f33f660f56dd21dce18044275e746374b1b6696e27d8868L22-R24) and use the hook instead?

Reply:
It seems cleaner to (1) open a PR with the new model-indexes hook against master and to merge it there. After that, (2) merge master back into `metrics-v2` and (3) rebase this branch on top of the updated metrics-v2.

That way this PR would not even touch model-indexes code, which would make the diff much more straightforward.

Either way is fine with me. Just listing options here, but will do whatever reviewers think is best.

Reply:
FWIW, model-indexes shouldn't be an entity - it's just that our data fetching/caching options are kind of limited, it was easiest to try to shove it into that framework. We're currently working establishing some nice easy patterns for fetching and caching non-entities that should make this a lot cleaner.

In the mean time, yes, absolutely update the API types in master if they no longer reflect the reality of what the backend is serving. I'd double check with a BE engineer to see what happened to `generation` and `state_changed_at`